src/txn/txn_recover.c

```diff
@@ -206,7 +209,12 @@ __txn_op_apply(WT_RECOVERY *r, WT_LSN *lsnp, const uint8_t **pp, const uint8_t *
         if (stop != NULL)
             stop->set_key(stop, stop_recno);
 
-        WT_TRET(session->iface.truncate(&session->iface, NULL, start, stop, NULL));
+        if (start == NULL && stop == NULL)
+            WT_TRET(
+              session->iface.truncate(&session->iface, r->files[fileid].uri, NULL, NULL, NULL));
+        else
+            WT_TRET(session->iface.truncate(&session->iface, NULL, start, stop, NULL));
```

Comment:
This change deserves a comment. The original code never called truncate with the URI.